### ISSUE TITLE
自训练rec模型acc一直为0,识别的是数字+英文字符+部分特殊字符。

### ISSUE THREAD

**Issue by GanWang** [PaddlePaddle/PaddleOCR#4972]
自训练rec模型acc一直为0,识别的是数字+英文字符+部分特殊字符。
[2021/12/20 07:35:20] root INFO: epoch: [44/100], iter: 2910, lr: 0.000307, loss: 25.555702, acc: 0.000000, norm_edit_dis: 0.097737, reader_cost: 0.00251 s, batch_cost: 100.58962 s, samples: 2560, ips: 2.54499
[2021/12/20 07:51:41] root INFO: epoch: [44/100], iter: 2920, lr: 0.000306, loss: 25.563505, acc: 0.000000, norm_edit_dis: 0.098403, reader_cost: 0.00241 s, batch_cost: 98.07543 s, samples: 2560, ips: 2.61024
[2021/12/20 08:08:01] root INFO: epoch: [44/100], iter: 2930, lr: 0.000305, loss: 25.127672, acc: 0.000000, norm_edit_dis: 0.100216, reader_cost: 0.00249 s, batch_cost: 97.90185 s, samples: 2560, ips: 2.61486
[2021/12/20 08:24:19] root INFO: epoch: [44/100], iter: 2940, lr: 0.000304, loss: 25.067665, acc: 0.000000, norm_edit_dis: 0.100262, reader_cost: 0.00231 s, batch_cost: 97.80230 s, samples: 2560, ips: 2.61753

**Comment by tink2123** [comment]
看学习率比较小，前期收敛可能比较慢。可以再多训一段时间观察loss下降情况，正常下降的话就没有问题。如果想加快收敛可以加载相应的预训练模型试试。

**Comment by GanWang** [comment]
> 

[2021/12/24 07:58:01] root INFO: epoch: [64/100], iter: 4260, lr: 0.000154, loss: 25.556374, acc: 0.000000, norm_edit_dis: 0.099754, reader_cost: 0.00260 s, batch_cost: 227.82876 s, samples: 2560, ips: 1.12365
[2021/12/24 08:36:01] root INFO: epoch: [64/100], iter: 4270, lr: 0.000153, loss: 25.664379, acc: 0.000000, norm_edit_dis: 0.100387, reader_cost: 0.00251 s, batch_cost: 227.90265 s, samples: 2560, ips: 1.12329
[2021/12/24 09:14:16] root INFO: epoch: [64/100], iter: 4280, lr: 0.000152, loss: 25.680859, acc: 0.000000, norm_edit_dis: 0.101382, reader_cost: 0.00240 s, batch_cost: 229.43829 s, samples: 2560, ips: 1.11577
loss在25附近就不动了，acc一直就是0

**Comment by GanWang** [comment]
使用的预训练模型时en_number_mobile_v2.0_rec_slim_train，配置文件是rec_en_number_lite_train.yml，识别的dict如下：0
1
2
3
4
5
6
7
8
9
a
b
c
d
e
f
g
h
i
j
k
l
m
n
o
p
q
r
s
t
u
v
w
x
y
z
A
B
C
D
E
F
G
H
I
J
K
L
M
N
O
P
Q
R
S
T
U
V
W
X
Y
Z
[
]
-
!
#
.
$
(
)
/
'
:
%
"
´
&
?
É
@
;
=
+
详细的配置文件：rec_en_number_lite_train.yml如下：
Global:
  use_gpu: False
  epoch_num: 100
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec_en_number_lite
  save_epoch_step: 2
  # evaluation is run every 2000 iterations after the 0th iteration
  eval_batch_step: [0, 2000]
  # if pretrained_model is saved in static mode, load_static_weights must set to True
  cal_metric_during_train: True
  pretrained_model: D:/Wanggan_Project/tensorflow/study_test/paddle_ocr/PaddleOCR2.3/pretrain_models/en_number_mobile_v2.0_rec_slim_train/best_accuracy
  checkpoints: 
  save_inference_dir:
  use_visualdl: False
  infer_img:
  # for data or label process
  character_dict_path: D:/Wanggan_Project/tensorflow/study_test/paddle_ocr/PaddleOCR2.3/ppocr/utils/dict/carnum_dict.txt
  character_type: ch
  max_text_length: 25
  infer_mode: False
  use_space_char: False


Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0005
    #0.001  0.000991
  regularizer:
    name: 'L2'
    factor: 0.00001

Architecture:
  model_type: rec
  algorithm: CRNN
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: small
    small_stride: [1, 2, 2, 2]
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 48
  Head:
    name: CTCHead
    fc_decay: 0.00001

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: E:/TXSJ/image/files_2twoall/imgs_rec_allnum/
    label_file_list: ["E:/TXSJ/image/files_2twoall/imgs_label_rec_allnum_path.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - RecAug: 
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 256
    drop_last: True
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: E:/TXSJ/image/files_2twoall/imgs_rec_allnum/
    label_file_list: ["E:/TXSJ/image/files_2twoall/imgs_label_rec_allnum_eval_path.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 256
    num_workers: 1
使用的样本是icdar2015加自己标注的样本13000张，训练loss在25就下去了，acc一直为0，是不是  Head:
    name: CTCHead
    fc_decay: 0.00001这个不对，修改了dict之后Head没修改？

**Comment by Liusdaooo** [comment]
我和 楼主有一样的 问题，我下载的预训练模型是ch_ppocr_server_v2.0_rec_train，在配置文件时，也是配置的ch_PP-OCRv2_rec.yml，但是，在训练文本识别的模型时，效果如下：
![效果图](https://github.com/user-attachments/assets/5adb6dc1-ef5d-4274-a583-06d7bd3c833e)
acc一直为0.00000
这个问题是怎么回事呢？
ch_PP-OCRv2_rec.yml配置文件如下：
![配置文件1](https://github.com/user-attachments/assets/12aba62c-c443-4369-bb24-c3d76bbb9f46)
![配置文件2](https://github.com/user-attachments/assets/86ba429a-fc6b-49e1-aee0-14a6a666030a)
![配置文件3](https://github.com/user-attachments/assets/31494ed5-3a1c-4034-a618-fff11583af8f)
![配置文件4](https://github.com/user-attachments/assets/c4723b32-d72c-4a5f-ae08-83af07461258)
数据集展示如下：
![数据集效果](https://github.com/user-attachments/assets/2a9d725a-f68b-4c3e-9d76-eac43bf71559)
